### DISCUSSION TITLE
Unable to run spark3.3.0 on airflow 1.10.14

### DISCUSSION THREAD

**Discussion by tejask8689** [apache/airflow#25955] [Q&A]
Hi,
I am working in a production environment, so I don't have access to attach the screenshots. We use GIT, Kubernetes, and AIRFLOW to run our pyspark jobs. So once the snapshot image is created in GIT, we use that image as a parameter in the bash command and trigger our pyspark jobs in airflow. The snapshot image which is made in GIT is using the spark3.3.0 version/spark2.4.1 version, we are triggering our pyspark jobs in airflow using the spark2.4.1 version and our airflow is running on the 1.10.14 version.

So when we use the snapshot image of the spark2.4.1 version we are not getting any issues. But when we are using the spark3.3.0 version the jobs are getting executed perfectly but they are going to up-for-retry state and retrying for 4 iterations and tasks are failing.

What could be the issue?

**Comment by potiuk** [comment]
You will not get much help here but likely it's dependency issue. Airflow 1.10 is end of line for way more than a year now and almost everyone migrated out to Airlfow 2 (see the last survey https://airflow.apache.org/blog/airflow-survey-2022/ ) which was done on 2021 and we are already almost close to the end of 2022 and we know for a fact that everyone who did not migrate, has already migrated.

So you are very close to situation described here:

![image](https://user-images.githubusercontent.com/595491/187008802-a2824abe-ee7a-4671-8e6e-2070f67607fe.png)

I am afraid the best answer you might get here is "Migrate ASAP".  Otherwise you will have to deal with simillar problems completely on your own.
